app/response_processor.py

```diff
@@ -108,19 +144,55 @@ def make_receipt(self, decrypted_json):
                     'user_id': decrypted_json['metadata']['user_id']
                 }
             }
-        except KeyError as e:
-            self.logger.error(
-                "Unsuccesful publish, missing key values", error=e)
+        except KeyError:
+            self.logger.exception("Unsuccesful publish, missing key values")
             raise QuarantinableError
 
+        return receipt_json
+
+    def _requires_dap_processing(self, decrypted_json):
+        if decrypted_json.get("survey_id") == "lms":
+            self.logger.info("LMS survey, sending to DAP", survey_id=decrypted_json.get("survey_id"))
+            return True
+        return False
+
+    def make_dap_data(self, decrypted_json):
+        self.logger.info("Creating dap data")
+
         try:
-            case_id = decrypted_json['case_id']
-            receipt_json['case_id'] = case_id
+            description = "{} survey response for period {} sample unit {}".format(
+                decrypted_json['survey_id'],
+                decrypted_json['collection']['period'],
+                decrypted_json['metadata']['ru_ref'])
+            dap_json = {
+                'version': 'v1',
+                'files': [{
+                    'name': '{}.json'.format(decrypted_json['tx_id']),
+                    'URL': '{}/{}'.format(settings.SDX_RESPONSES_URL, decrypted_json['tx_id'])
+                }],
+                'sensitivity': 'High',
+                'sourceName': settings.DAP_SOURCE_NAME,
+                'manifestCreated': datetime.utcnow().isoformat(),
+                'description': description,
+                'iterationL1': decrypted_json['collection']['period'],
+                'dataset': decrypted_json['survey_id'],
+                'schemaversion': "1"
+            }
         except KeyError:
-            # Don't do anything, as this survey originated from RRM
-            logger.debug("Received an rrm survey")
+            self.logger.exception("Unsuccesful publish, missing key values")
+            raise QuarantinableError
+        response = self.remote_call('{}/{}'.format(settings.SDX_RESPONSES_URL, decrypted_json['tx_id']))
+        try:
+            self.response_ok(response)
+        except ClientError:
+            self.logger.error("Survey retrieval failed. Quarantining Survey.")
+            raise QuarantinableError
 
-        return receipt_json
+        dap_json['files'][0]['sizeBytes'] = response.headers['Content-Length']
+        dap_json['files'][0]['md5sum'] = response.headers['Content-MD5']
```

Comment:
Can the call to store be done first, so that when constructing the dap response the size and md5sum can be added at the same time as name and URL?

Reply:
Huh...Not sure why I didn't think of doing that.  Good idea